localstack/testing/pytest/fixtures.py

```diff
@@ -278,6 +222,29 @@ def s3_bucket(s3_create_bucket, aws_client) -> str:
     return s3_create_bucket(**kwargs)
 
 
+@pytest.fixture
+def s3_empty_bucket(aws_client):
+    """
+    Returns a factory that given a bucket name, deletes all objects and deletes all object versions
+    """
+    # Boto resource would make this a straightforward task, but our internal client does not support Boto resource
+    # FIXME: this won't work when bucket has more than 1000 objects
+    def factory(bucket_name: str):
+        response = aws_client.s3.list_objects_v2(Bucket=bucket_name)
+        objects = [{"Key": obj["Key"]} for obj in response["Contents"]]
+        aws_client.s3.delete_objects(Bucket=bucket_name, Delete={"Objects": objects})
+
+        response = aws_client.s3.list_object_versions(Bucket=bucket_name)
+        object_versions = [
+            {"Key": obj["Key"], "VersionId": obj["VersionId"]}
+            for obj in response.get("Versions", [])
+        ]
+        if object_versions:
+            aws_client.s3.delete_objects(Bucket=bucket_name, Delete={"Objects": object_versions})
+
+    yield factory
```

Comment:
This new fixture replaces the functionality offered by the Boto S3 Resource

Reply:
I'd probably prefer this to be a normal utility over a fixture but should be ok to leave as-is for now :+1: